python/python/bystro/ancestry/listener.py

```diff
@@ -0,0 +1,105 @@
+"""Provide a worker for the ancestry model."""
+import argparse
+import logging
+from pathlib import Path
+
+from ruamel.yaml import YAML
+
+from bystro.ancestry.ancestry_types import AncestryResponse, AncestrySubmission
+from bystro.beanstalkd.messages import BaseMessage, CompletedJobMessage, SubmittedJobMessage
+from bystro.beanstalkd.worker import ProgressPublisher, QueueConf, get_progress_reporter, listen
+
+logger = logging.getLogger()
+
+ANCESTRY_TUBE = "ancestry"
+
+
+class AncestryJobData(BaseMessage, frozen=True):
+    """Wrap an AncestrySubmission in a BaseMessage for beanstalk."""
+
+    ancestry_submission: AncestrySubmission
+
+
+class AncestryJobCompleteMessage(CompletedJobMessage, frozen=True, kw_only=True):
+    """Wrap an AncestryResponse in a CompletedJobMessage for beanstalk."""
+
+    results: AncestryResponse
+
+
+def _load_queue_conf(queue_conf_path: str) -> QueueConf:
+    with Path(queue_conf_path).open(encoding="utf-8") as queue_config_file:
+        raw_queue_conf = YAML(typ="safe").load(queue_config_file)
+    beanstalk_conf = raw_queue_conf["beanstalkd"]
+    return QueueConf(addresses=beanstalk_conf["addresses"], tubes=beanstalk_conf["tubes"])
+
+
+def _infer_ancestry(
+    ancestry_submission: AncestrySubmission, _publisher: ProgressPublisher
+) -> AncestryResponse:
+    """Run an ancestry job."""
+    # TODO: main ancestry model logic goes here.  Just stubbing out for now.
+    logger.debug("Inferring ancestry for: %s", ancestry_submission)
+
+    # not doing anything with this reporter at the moment, we're
+    # simply threading it through for later.
+    _reporter = get_progress_reporter(_publisher)
+    vcf_path = ancestry_submission.vcf_path
+    return AncestryResponse(vcf_path=vcf_path, results=[])
+
+
+async def handler_fn(
```

Comment:
This doesn't need to be an async function, necessarily, but feel free to leave it. We may need to update beanstalkd to handle non-async functions.

Reply:
With Ray, the reason to have async is if the main function is doing a bunch of non-blocking work.

Reply:
Yeah I'd love to make this an eager function if we can get away with it.  When I remove the async, though, I get the following error when running the script and handling a message:

```
Traceback (most recent call last):
  File "/Users/patrickoneil/bystro/python/python/bystro/beanstalkd/worker.py", line 153, in listen
    res = asyncio.get_event_loop().run_until_complete(handler_fn(publisher, job_data))
  File "/opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 628, in run_until_complete
    future = tasks.ensure_future(future, loop=self)
  File "/opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/tasks.py", line 615, in ensure_future
    return _ensure_future(coro_or_future, loop=loop)
  File "/opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/tasks.py", line 630, in _ensure_future
    raise TypeError('An asyncio.Future, a coroutine or an awaitable '
TypeError: An asyncio.Future, a coroutine or an awaitable is required
```

Is there a way to use the existing `listen` API with an eager handler_fn?